src/core/src/package_managers/YumPackageManager.py

```diff
@@ -347,26 +386,249 @@ def get_package_size(self, output):
+    def disable_auto_update_on_reboot(self, command):
+        self.composite_logger.log_debug("Disabling auto update on reboot using command: " + str(command))
+        code, out = self.env_layer.run_command_output(command, False, False)
+        self.composite_logger.log_debug(" - Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+
+        if code != 0:
+            self.composite_logger.log('[ERROR] Command invoked: ' + command)
+            self.telemetry_writer.write_execution_error(command, code, out)
+            error_msg = 'Unexpected return code (' + str(code) + ') on command: ' + command
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.OPERATION_FAILED)
+            raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+
+        self.composite_logger.log_debug("Auto update on reboot disabled")
+
+    def is_auto_update_service_installed(self, install_check_cmd):
+        """ Checks if the auto update service is enable_on_reboot on the VM """
+        self.composite_logger.log_debug("Checking if auto update service is installed...")
+        code, out = self.env_layer.run_command_output(install_check_cmd, False, False)
+        self.composite_logger.log_debug(" - Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+        if len(out.strip()) > 0 and code == 0:
+            self.composite_logger.log_debug("Auto OS update is installed on the machine")
+            return True
+        else:
+            self.composite_logger.log_debug("Auto OS update is NOT installed on the machine")
+            return False
+
+    def install_auto_update_service(self, command):
+        self.composite_logger.log_debug("Installing auto updates using command: " + str(command))
+        code, out = self.env_layer.run_command_output(command, False, False)
+        if code != 0:
+            self.composite_logger.log('[ERROR] Package manager was invoked using: ' + command)
+            self.composite_logger.log_warning(" - Return code from package manager: " + str(code))
+            self.composite_logger.log_warning(" - Output from package manager: \n|\t" + "\n|\t".join(out.splitlines()))
+            self.telemetry_writer.write_execution_error(command, code, out)
+            error_msg = 'Unexpected return code (' + str(code) + ') from package manager on command: ' + command
+            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+            raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+        else:
+            self.composite_logger.log_debug("\n\n==[SUCCESS]===============================================================")
+            self.composite_logger.log_debug(" - Return code from package manager: " + str(code))
+            self.composite_logger.log_debug(" - Output from package manager: \n|\t" + "\n|\t".join(out.splitlines()))
```

Comment:
Single line

Reply:
Same as above, this code doesn't exist now